core/internal/monitor/monitor.go

```diff
@@ -447,6 +455,39 @@ func (sm *SystemMonitor) monitorResource(resource Resource) {
 
 }
 
+// ShouldCaptureSamplingError checks if a resource sampling error should be sent to Sentry.
+//
+// Use to filter out expected/transient failures. Keep this intentionally small and specific.
+func ShouldCaptureSamplingError(err error) bool {
```

Comment:
style: a switch is neater

```go
switch {
case strings.Contains(msg, "connection refused"):
	return false

// Platforms without netstat (gopsutil may shell out on some OSes).
case errors.Is(err, exec.ErrNotFound):
	return false
case strings.Contains(msg, "executable file not found") && strings.Contains(msg, "netstat"):
	return false

// Windows sporadic low-level API failures.
case strings.Contains(msg, "Incorrect function."):
	return false

...

default:
	return true
}
```

Use blank lines to break up groups of disjunctions (e.g. the netstat check is cleaner as two case statements rather than one wrapped one using `||`)